packages/playwright/src/reporters/base.ts

```diff
@@ -254,11 +254,14 @@ export class BaseReporter implements ReporterV2 {
   }
 
   private _printMaxFailuresReached() {
-    if (!this.config.maxFailures)
+    let maxFailures = this.config.maxFailures;
+    maxFailures = Number.isInteger(maxFailures) ? maxFailures : Math.ceil(maxFailures * this.totalTestCount);
+
+    if (!maxFailures)
       return;
-    if (this._failureCount < this.config.maxFailures)
+    if (this._failureCount < maxFailures)
       return;
-    console.log(colors.yellow(`Testing stopped early after ${this.config.maxFailures} maximum allowed failures.`));
+    console.log(colors.yellow(`Testing stopped early after ${maxFailures} maximum allowed failures.`));
```

Comment:
Here in case of specifying the value as float number will be displayed only calculated value relative to total tests. Maybe in this case I should add also real specified `maxFailures` in config/cli.